src/big/big_float.cr

```diff
@@ -51,16 +83,36 @@ struct BigFloat < Float
     LibGMP.mpf_cmp(self, other)
   end
 
-  def <=>(other : Float)
+  def <=>(other : BigInt)
+    LibGMP.mpf_cmp_z(self, other)
+  end
+
+  def <=>(other : Float32 | Float64)
     LibGMP.mpf_cmp_d(self, other.to_f64)
   end
 
   def <=>(other : Int::Signed)
-    LibGMP.mpf_cmp_si(self, other.to_i64)
+    if LibGMP::Long == Int64
+      LibGMP.mpf_cmp_si(self, other.to_i64)
+    elsif other.is_a?(Int8 | Int16 | Int32)
+      LibGMP.mpf_cmp_si(self, LibGMP::Long.new(other))
```

Comment:
We need `LibGMP::Long` here (it is Int32 or Int64 for now). How to cast to it without `new`?